workflow/metrics/metrics.go

```diff
@@ -89,28 +91,41 @@ func (m *Metrics) allMetrics() []prometheus.Metric {
 }
 
 func (m *Metrics) WorkflowAdded(key string, phase v1alpha1.NodePhase) {
-	if m.workflows[key] {
```

Comment:
Could you simplify to use `sync.Mutex` for and `defer`. I think the benefit of simpler code would outweight the reduced efficiency.